app/components/layer-menu-item.js

```diff
@@ -1,47 +1,35 @@
 import Component from '@ember/component';
-import { alias } from '@ember/object/computed';
 import { inject as service } from '@ember/service';
-import { computed } from '@ember/object'; // eslint-disable-line
 
 export default Component.extend({
-  registeredLayers: service(),
-  metrics: service(),
-
-  visible: alias('layer.visible'),
-  // tagName: 'li',
-  classNames: 'layer-menu-item',
-
-  layer: computed('for', 'registeredLayers.layers.@each', function() {
-    const { for: layerId, 'registeredLayers.layers': layers = [] } = this.getProperties('for', 'registeredLayers.layers');
-
-    return layers.findBy('config.id', layerId);
-  }),
-
-  title: alias('layer.config.title'),
-
-  legendIcon: alias('layer.config.legendIcon'),
-
-  legendColor: alias('layer.config.legendColor'),
+  init(...args) {
+    this._super(...args);
+
+    const recordIdentifier = this.get('for');
+    const foundRecord = this.get('store').peekRecord('layer-group', recordIdentifier);
+    if (foundRecord) {
+      this.set('model', foundRecord);
+    }
+  },
 
-  titleTooltip: alias('layer.config.titleTooltip'),
+  store: service(),
 
+  for: '',
```

Comment:
style nit: line break between properties, especially before the `actions` object